sql/parser/eval.go

```diff
@@ -1179,35 +1181,39 @@ func ParseDate(s DString) (DDate, error) {
 }
 
 // ParseTimestamp parses the timestamp.
-func ParseTimestamp(s DString) (DTimestamp, error) {
+func (ctx EvalContext) ParseTimestamp(s DString) (DTimestamp, error) {
 	str := string(s)
 	t, err := time.Parse(dateFormat, str)
 	if err == nil {
 		return DTimestamp{Time: t}, nil
 	}
-	t, err = time.Parse(timestampFormat, str)
+	loc := time.UTC
+	if ctx.Offset != 0 {
+		loc = time.FixedZone("", int(ctx.Offset/int64(time.Second)))
```

Comment:
Perhaps `EvalContext.Offset` should just be seconds. I would propagate this up to the session state as well.